app/Services/Model/Imp/SummitRegistrationInvitationService.php

```diff
@@ -627,17 +630,37 @@ function ($summit, $flow_event, $invitation_id, $test_email_recipient, $announce
                             return $invitation;
                         });
 
+                        $add_excerpt = false;
+
                         // send email
-                        if ($flow_event == InviteSummitRegistrationEmail::EVENT_SLUG && !is_null($invitation))
+                        if ($flow_event == InviteSummitRegistrationEmail::EVENT_SLUG && !is_null($invitation)) {
                             InviteSummitRegistrationEmail::dispatch($invitation, $test_email_recipient);
-                        if ($flow_event == ReInviteSummitRegistrationEmail::EVENT_SLUG && !is_null($invitation))
+                            $add_excerpt = true;
+                        }
+
+                        if ($flow_event == ReInviteSummitRegistrationEmail::EVENT_SLUG && !is_null($invitation)) {
                             ReInviteSummitRegistrationEmail::dispatch($invitation, $test_email_recipient);
+                            $add_excerpt = true;
+                        }
+
+                        if ($add_excerpt) {
+                            EmailExcerpt::add(
```

Comment:
could you refactor this https://github.com/OpenStackweb/summit-api/blob/206972e67573d795b14311244a40a8c08d5eefa2/app/Services/Model/Imp/Traits/ParametrizedSendEmails.php#L200 so accepts a custom callback to generate the success dispatch instead of doing this ?